### ISSUE TITLE
Redis High Ram Consumption

### ISSUE THREAD

**Issue by GiuseppeLaurenza** [LuatixHQ/opencti#1898]
Dear all, after update my docker swarm installation to 5.1.4, i have redis container that use quite half of the available RAM
In the past, on chat someone suggest me to put REDIS__TRIMMING=50000 in compose file and it solved the problem, but now it appears that it doesn't produce any effect
On platform and redis log, I can't found anything strange
Have you got any ideas?
Thanks in advance

**Comment by gfmellado** [comment]
We're having the same issue in our docker swarm installation, redis allocates almost 31GB RAM on a 32GB RAM host and it doesn't work if we limit the memory either via redis option or docker limits.
Is there a posible solution to this issue that we're missing?

**Comment by SamuelHassine** [comment]
Hello @GiuseppeLaurenza, @gfmellado,

Did you consider using the trimming parameter? To avoid Redis to grow indefinitely with your platform.

Kind regards,
Samuel

**Comment by SamuelHassine** [comment]
Hello,

Link: https://luatix.notion.site/redis-trimming-aeb7fd60008f45d79885888e120d53d4.

Kind regards,
Samuel

**Comment by GiuseppeLaurenza** [comment]
@SamuelHassine I forgot to close the issue, as I say at first post, I tried with the parameter, but it not solve the problem. In the community someone (maybe you, I do not remember) suggest to open redis and manually trim the queue for the first time. It solved the issue and currently I have no problem with it

**Comment by sanctifie** [comment]
hi everyone i have the same problem with the ram! does this link no longer work?

https://luatix.notion.site/redis-trimming-aeb7fd60008f45d79885888e120d53d4.

**Comment by sanctifie** [comment]
hi everyone i have the same problem with the ram! does this link no longer work?
https://luatix.notion.site/redis-trimming-aeb7fd60008f45d79885888e120d53d4.

**Comment by kmcmahon1959** [comment]
I am having the same issue.  On my development instance I set REDIS__TRIMMING on the platform container to 2000000 (the stream.opencti had over 5M entries).  This did not seem to have any effect.  I then manually ran XTRIM on the redis container to remove a bunch of entries; and it seemed to run well, but still used up all the memory.  Then the redis container restarted and is now back to 5M+ entries in the stream.opencti key. ???
